### ISSUE TITLE
Mismatch between External Secrets Operator version and CRDs in the FluxCD documentation

### ISSUE THREAD

**Issue by roy-hardin** [external-secrets/external-secrets#3885]
**Describe the bug**
The documentation for the External Secrets Operator on the [Examples → FluxCD](https://external-secrets.io/latest/examples/gitops-using-fluxcd/) page explains how to install External Secrets using Flux.

In the “_Deploy the operator_” section, the example code shows how to use the Helm chart with version `0.9.4`. However, in the section “_Creating the references to repositories_,” the steps for connecting the `GitRepository` do not include a tag that matches the chart version.

This causes the Operator to be installed at version 0.9.4, but the Custom Resource Definitions (CRDs) to be installed from the `HEAD`  of the Git repository. As a result, the error described in issue #3882 occurred (which has already been fixed). Without explicitly specifying a tag for the GitRepository, there may be a risk of incompatibilities between the operator and the CRDs.

**To Reproduce**
1. Follow the instructions in Examples → FluxCD, using version 0.9.4 of the chart.
2. Create references to the Git repository as described, without specifying a tag corresponding to the chart version.
3. Observe that the operator version 0.9.4 is installed, but the CRDs from the HEAD are applied.
4. Observe that the CustomResourceDefinitions will be annotated with `controller-gen.kubebuilder.io/version: v0.16.2`. 

**Expected behavior**
The installed version of the CRDs should match the installed version of the Operator (0.9.4).

The documentation could explicitly instruct how to specify the correct tag in the repository reference to avoid version mismatches.

```yaml
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: GitRepository
metadata:
  name: external-secrets
  namespace: flux-system
spec:
  interval: 10m
  ref:
    branch: main
    tag: v0.9.4
```

**Additional context**

Additional proposals:

Update version numbers of the Helm Chart and the Flux API in YAML code:

1. Replace tag `v0.9.4` with `v0.10.2` and chart version `0.9.4` with `0.10.2`.
2. Replace `apiVersion: source.toolkit.fluxcd.io/v1beta1` with `apiVersion: source.toolkit.fluxcd.io/v1`
3. Replace `apiVersion: helm.toolkit.fluxcd.io/v2beta1` with `apiVersion: helm.toolkit.fluxcd.io/v2`
4. Replace `apiVersion: kustomize.toolkit.fluxcd.io/v1beta2` with `apiVersion: kustomize.toolkit.fluxcd.io/v1`

**Comment by Skarlso** [comment]
Ah nice find! That's why your CRDs where from HEAD.

**Comment by buehren** [comment]
Thank you very much for posting this!

Glad I found this after hours of investigating what might be wrong with Flux.

**Comment by buehren** [comment]
BTW: The changes listed under "Additional proposals" also work for me.

But I skipped version `0.10.2` and used `0.10.3`.
* That did not work on my playground cluster - external-secrets-crds remained in `Reconciliation in progress` even after `Helm install succeeded for release external-secrets/external-secrets.v1 with chart external-secrets@0.10.3`.
* I reverted to `0.9.4` which works. I might try an update later to see if the newer version will also work. Maybe it already does for the rest of you.
